File: Algorithms/p055_Jump_Game/p055_Jump_Game_1.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int min_reachable = nums.size()-1;
        for(int i=nums.size()-2;i>=0;--i){
            if(i+nums[i]>=min_reachable) min_reachable = i;
        }
        return min_reachable==0;
    }
};

File: Algorithms/p055_Jump_Game/p055_Jump_Game_1.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        min_reachable = len(nums)-1
        for i in range(len(nums)-2,-1,-1):
            if i+nums[i]>=min_reachable:
                min_reachable = i
        return min_reachable==0
